planner/core/find_best_task.go

```diff
@@ -516,13 +521,9 @@ func (ds *DataSource) convertToIndexScan(prop *property.PhysicalProperty, candid
 		is.KeepOrder = true
 		is.addPushedDownSelection(cop, ds, prop.ExpectedCnt, path)
 	} else {
-		expectedCnt := math.MaxFloat64
-		if prop.IsEmpty() {
-			expectedCnt = prop.ExpectedCnt
-		} else {
-			return invalidTask, nil
-		}
-		is.addPushedDownSelection(cop, ds, expectedCnt, path)
+		// prop.IsEmpty() would always return true when coming to here,
+		// so we can just use prop.ExpectedCnt as parameter of addPushedDownSelection.
+		is.addPushedDownSelection(cop, ds, prop.ExpectedCnt, path)
```

Comment:
We can further more extract `addPushedDownSelection` out of the if-else branch now?

Reply:
done.